# Phylogenetic Analysis of Clustered MSA Sequences

This folder contains scripts and outputs related to the phylogenetic analysis of clustered protein sequences. 

## Phylogenetic Tree Construction

The phylogenetic tree was generated using [IQ-TREE](https://iqtree.github.io/) with the following command:

```
iqtree -s MSA_80cluster.fas -nt AUTO -bb 1000 -pre MSA_80cluster
```

## Visualization in iTOL

To visualize the phylogenetic tree with metadata annotations:

1. Upload `MSA_80cluster_cleaned.treefile` to [iTOL](https://itol.embl.de/).
2. Import `dataset_colorstrip.txt` or `dataset_colorstrip_all.txt` as annotations.


## Description of Files

| File                                 | Description                                                                 |
|--------------------------------------|-----------------------------------------------------------------------------|
| `MSA_80cluster.fas`                  | Input FASTA file containing sequences for IQ-TREE.   |
| `MSA_80cluster.treefile`            | Newick-format tree generated by IQ-TREE. |
| `MSA_80cluster_cleaned.treefile`    | Tree with cleaned/standardized leaf names, processed by `clean_tree.py`. |
| `MSA_80cluster_with_blast_names.csv`| Metadata for each sequence including `Common_taxon`, `Common name`, and `BLAST name`. Used for annotations. `Common name` is derived from `BLAST name`, and is used for simplicity in visualization.|
| `dataset_colorstrip.txt`           | iTOL annotation file highlighting the top 5 most frequent common names (Figure 1) |
| `dataset_colorstrip_all.txt`       | iTOL annotation file with all unique common names, sorted by frequency (see Appendix) |
| `clean_tree.py`                     | Python script that: <br> - Cleans leaf names for iTOL <br> - Applies color mapping based on metadata <br>|

